pkg/cache/tas_flavor_snapshot.go

```diff
@@ -422,7 +429,16 @@ func (s *TASFlavorSnapshot) findTopologyAssignment(
 		return nil, fmt.Sprintf("no requested topology level: %s", *key)
 	}
 	// phase 1 - determine the number of pods which can fit in each topology domain
-	s.fillInCounts(requests, assumedUsage, simulateEmpty, append(podSetTolerations, s.tolerations...))
+	err := s.fillInCounts(
+		requests,
+		assumedUsage,
+		simulateEmpty,
+		append(podSetTolerations, s.tolerations...),
+		podSetNodeSelectors,
+	)
+	if err != nil {
+		return nil, fmt.Sprintf("invalid node selectors: %s, reason: %s", podSetNodeSelectors, err)
+	}
```

Comment:
@mimowo Is this how you wanted to return and propagate the error?

Reply:
Pretty much yes. 

Non-blocking nit: I think we can yet have less code if call ValidatedSelectorFromSet outside fillInCounts, and only pass already compiled selector. This way we can save one layer of propagating the errors.